Vynke comments 01 (#59)
ietf-homenet-wg/ietf-homenet-hna: draft-ietf-homenet-front-end-naming-delegation.md

By accepting the DS RR, the DM commits to advertise the DS to the parent zone.
On the other hand if the DM does not have the capacity to advertise the DS to the parent  zone, it indicates this by refusing the update to the DS RR.

## Information to set up the Synchronization Channel {#sec-sync-info}

The HNA works as a hidden primary authoritative DNS server, while the DM works like a secondary.
As a result, the HNA must provide the IP address the DM should use to reach the HNA.

If the HNA detects that it has been renumbered, then it MUST use the Control Channel to update the DOI with the new IPv6 address it has been assigned.

> The Synchronization Channel will be set between the new IPv6 address and the IP address of the DM.

Reviewer: I suspect we should probably only say the new IP instead of the new IPv6

Author: Do we support Synchronization over IPv4?

Reviewer: The DNS mechanism does not mandate IPv6, so I have been asked to not prevent IPv4 and as far as I can see we do support IPv4.

> The Synchronization Channel will be set between the new IPv6 address and the IP address of the DM.

Author: ```suggestion
The Synchronization Channel will be set between the new IPv6 (and IPv4) address and the IP address of the DM.
```

AXFR enables the response to contain any type of RRsets.

To retrieve the necessary information to build the Public Homenet Zone, the HNA MUST send a DNS request of type AXFR associated with the Registered Homenet Domain.

The zone that is returned by the DM is used by the HNA as a template to build its own zone.

> The zone template MUST contain a RRset of type SOA, one or multiple RRset of type NS and zero or more RRset of type A or AAAA (if the NS are in-bailiwick {{!RFC8499}}).

Reviewer: I am wondering if we should add the TTL values accepted by the DOI.  This is clarified in the #zonetemplate section, so that is not essential.

Author: The TTL values would be part of any RR received in the template, but maybe you mean, that they should be taken as a recommendation?

Reviewer: Yes I meant to take it as a recommendation.

> The zone template MUST contain a RRset of type SOA, one or multiple RRset of type NS and zero or more RRset of type A or AAAA (if the NS are in-bailiwick {{!RFC8499}}).

Author: ```suggestion
The zone template MUST contain a RRset of type SOA, one or multiple RRset of type NS and zero or more RRset of type A or AAAA (if the NS are in-bailiwick {{!RFC8499}}).
The zone template will include Time To Live (TTL) values for each RR, and the HNA SHOULD take these as suggested maximum values, but MAY use lower values for operational reasons, such impending renumbering events.
```

This is handled using the normal zone transfer mechanism involving AXFR/IXFR.

Part of this zone update process involves the owner of the zone (the hidden primary, the HNA) sending a DNS Notify to the secondaries.
In this situation the only destination that is known by the HNA is the DM's Control Channel, and so DNS notifies are sent over the Control Channel, secured by TLS.

> However, DNS Notifies are not critical: they just cause the DM to use the Synchronization channel to do an SOA Query to detect any updates, and if there are some, then to transfer the zone.

Reviewer: "However, DNS Notifies are not critical" I was expecting the text that follows to say that other channels 53 may be used as a consequence that Notifies are not critical. I do not read we give this possibility and I am wondering why we do have the "however".  I am readin gin fact the paragraph as a note, that is an information.

Author: I don't think that a DNS Notify to port 53 of the DM would necessarily mean anything.  It might not lead to the Synchronization system.  We don't say that 53 has to be alive, only 853.

Reviewer: I agree. My only note was that I found strange "However: that seems to indicate something will follows that contradict what has just been said. That said, you know english better than I, so if "However" is fine, just ignore my comment.

> However, DNS Notifies are not critical: they just cause the DM to use the Synchronization channel to do an SOA Query to detect any updates, and if there are some, then to transfer the zone.

Author: ```suggestion
Please note that, DNS Notifies are not critical to normal operation, as the DM will be checking the zone regularly based upon SOA record comments.  DNS Notifies do speed things up as they cause the DM to use the Synchronization channel to immediately do an SOA Query to detect any updates.  If there are any changes then the DM immediately transfers the zone updates.
```

# Public Homenet Reverse Zone {#sec-reverse}

Public Homenet Reverse Zone works similarly to the Public Homenet Zone.

> The main difference is that ISP that provides the IPv6 connectivity is likely also the owner of the corresponding IPv6 reverse zone and administrating the Reverse Public Authoritative Servers.

Reviewer: Initially the draft was IPv6 focused and we changed it to IP as most of the mechanism were also applicable with IPv4. I am wondering if we should specify IPv6 here. I agree that only DHCPv6 have been defined and that it makes more sense with IPv6 than IPv4, so my comment is just for consistency.

Reviewer 2: Without any hat... reverse zone is mainly for IPv6 though. So, it makes sense to have an IPv6-specific text

Reviewer: I also agree here.
